### PR TITLE
feat(apig/signature): add new association resource

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Support a new APIG resource to bind the signature to the published APIs.

Notes: A published API can only has one signature.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support a new APIG resource to bind the signature to the published APIs.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccSignatureAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccSignatureAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccSignatureAssociate_basic
=== PAUSE TestAccSignatureAssociate_basic
=== CONT  TestAccSignatureAssociate_basic
--- PASS: TestAccSignatureAssociate_basic (519.78s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      519.835s
```

### CODE REVIEW

File: huaweicloud/services/apig/resource_huaweicloud_apig_signature_associate.go [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+		for _, val := range resp {
+			if val.BindId == bandId {
+				return resp, "PENDING", nil
+			}
+		}
+		return resp, "COMPLETED", nil
+	}
+}
+
+func unbindSignatureFromApis(ctx context.Context, client *golangsdk.ServiceClient, instanceId, signId string,
```
Comment (ShiChangkuo) [inline huaweicloud/services/apig/resource_huaweicloud_apig_signature_associate.go]: it seems that this function will unbind all of the publish ids, please check it.

Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_signature_associate.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccSignatureAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccSignatureAssociate_basic -timeout 360m -parallel 4
^[l=== RUN   TestAccSignatureAssociate_basic
=== PAUSE TestAccSignatureAssociate_basic
=== CONT  TestAccSignatureAssociate_basic
--- PASS: TestAccSignatureAssociate_basic (522.69s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      522.754s
```

**Pull request merged**